src/components/VNotificationBanner.vue

```diff
@@ -59,17 +58,25 @@ export default defineComponent({
     close: defineEvent(),
   },
   setup(props, { emit }) {
-    const i18n = useI18n()
-    const shouldShow = useStorage(`banner:show-${props.id}`, true)
-    const dismissBanner = () => (shouldShow.value = false)
+    const uiStore = useUiStore()
+
+    /**
+     * We only hide the enabled banner if there is a cookie.
+     */
+    const shouldShow = ref(!uiStore.isBannerDismissed(props.id))
```

Comment:
This is definitely just a nit about how to manage refs and store-derived data: Would it make sense to call `isBannerDismissed` inside of a computed callback? That way the `dismissBanner` function doesn't have to futz with updating the ref:

```ts
const shouldShow = computed(() => !uiStore.isBannerDismissed(props.id))
```